azext_iot/tests/test_iot_central_int.py

```diff
@@ -91,24 +93,138 @@ def test_central_device_twin_show_success(self):
-        self.cmd("iot central app validate-messages --app-id {} --to 1".format(APP_ID))
+        device_client.send_message(msg)
+
+        # Validate the messages
+        output = self._get_validate_messages_output(device_id, enqueued_time)
+
+        self._delete_device(device_id)
+        self._delete_device_template(template_id)
+
+        assert output
+        assert "Successfully parsed 1 message(s)" in output
+        assert "No errors detected" in output
+
+    def test_central_validate_messages_issues_detected(self):
+        expected_messages = []
+        (template_id, _) = self._create_device_template()
+        (device_id, _) = self._create_device(instance_of=template_id)
+        credentials = self._get_credentials(device_id)
+
+        device_client = helpers.dps_connect_device(device_id, credentials)
+
+        enqueued_time = utility.calculate_millisec_since_unix_epoch_utc() - 10000
+
+        # Invalid encoding
+        payload = {"Bool": True}
+        msg = Message(data=json.dumps(payload), content_type="application/json")
+        device_client.send_message(msg)
+        expected_messages.append(strings.invalid_encoding(""))
+
+        # Content type mismatch (e.g. non application/json)
+        payload = {"Bool": True}
+        msg = Message(data=json.dumps(payload), content_encoding="utf-8")
+        device_client.send_message(msg)
+        expected_messages.append(strings.content_type_mismatch("", "application/json"))
+
+        # Invalid type
+        payload = {"Bool": 123}
+        msg = Message(
+            data=json.dumps(payload),
+            content_encoding="utf-8",
+            content_type="application/json",
+        )
+        device_client.send_message(msg)
+        expected_messages.append(
+            strings.invalid_primitive_schema_mismatch_template("Bool", "boolean", 123)
+        )
+
+        # Telemetry not defined
+        payload = {"NotPresentInTemplate": True}
+        msg = Message(
+            data=json.dumps(payload),
+            content_encoding="utf-8",
+            content_type="application/json",
+        )
+        device_client.send_message(msg)
+        # this error is harder to build from strings because we have to construct a whole template for it
+        expected_messages.append(
+            "Following capabilities have NOT been defined in the device template '['NotPresentInTemplate']'"
+        )
+
+        # Invalid JSON
+        payload = '{"asd":"def}'
+        msg = Message(
+            data=payload, content_encoding="utf-8", content_type="application/json",
+        )
+        device_client.send_message(msg)
+        expected_messages.append(strings.invalid_json())
+
+        # Validate the messages
+        output = self._get_validate_messages_output(
+            device_id, enqueued_time, max_messages=len(expected_messages)
+        )
+
+        self._delete_device(device_id)
```

Comment:
nit/future iteration thing: Consider looking into usage of fixtures and/or usage of xunit style setup & teardown to make it easier and faster to setup and clean up after a test.

Reply:
Sounds good, I'll keep that in mind for next work items when they come down the pipe